### PR TITLE
Add Chinese YOLO research customization quickstart

### PR DESCRIPTION
**Pull request by wangjia15**
## Summary

- add a Chinese quickstart that maps the source tree and training call chain
- explain the smallest extension point for config, YAML, modules, trainers, loss, and data research
- include a reproducible baseline-to-export workflow and troubleshooting checklist
- link the guide from the Chinese README

## Validation

- 
px prettier@3.8.5 --print-width 120 --check README.zh-CN.md RESEARCH_QUICKSTART.zh-CN.md`
- git diff main...HEAD --check`

## Notes

Documentation-only change; no runtime code was modified.

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
Added a Chinese YOLO research customization quickstart and linked it from `README.zh-CN.md` to help researchers identify appropriate source-level extension points and follow a reproducible experimentation workflow.

### 📊 Key Changes
- Added `RESEARCH_QUICKSTART.zh-CN.md` with a source-tree map and training call-chain overview.
- Documented minimal customization paths for configuration, model YAML, custom modules, trainers, losses, data pipelines, and task components.
- Included editable installation, dataset setup, baseline training, controlled experiments, validation, export, and verification examples.
- Added research-oriented validation and troubleshooting checklists covering module registration, tensor shapes, imports, distributed training, pretrained weights, and backend export.
- Linked the new guide from the Chinese README documentation section.

### 🎯 Purpose & Impact
Provides Chinese-language guidance for reading and customizing the Ultralytics codebase while preserving existing runtime behavior; no Python runtime code was modified.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @wangjia15, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This automated message confirms your contribution was received, and an Ultralytics engineer will assist with the review. To ensure a seamless integration of your work, please review the following checklist:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_ — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

File: RESEARCH_QUICKSTART.zh-CN.md [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+
+    def preprocess_batch(self, batch):
+        """Reuse standard preprocessing before applying the intervention."""
+        batch = super().preprocess_batch(batch)
+        # 在这里加入确有必要的 batch 级处理
+        return batch
+
+
+model = YOLO("yolo26n.pt")
+model.train(data="path/to/dataset.yaml", trainer=ResearchTrainer)
```
Comment (UltralyticsAssistant) [inline RESEARCH_QUICKSTART.zh-CN.md:229]: 💡 **MEDIUM**: 此示例把 `ResearchTrainer` 定义在调用脚本的 `__main__` 中。单卡可用，但多卡训练时 `BaseTrainer` 会生成临时 DDP 脚本，并按 `trainer.__class__.__module__` 执行 `from __main__ import ResearchTrainer`；此时 `__main__` 已是临时脚本，导入会失败。请将 trainer 放入可导入的模块后再传入，或明确该示例仅适用于单卡，并在多卡排查项中说明。

File: RESEARCH_QUICKSTART.zh-CN.md [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+### 第一步：创建独立环境并以 editable 模式安装
+
+```bash
+git clone https://github.com/ultralytics/ultralytics.git
+cd ultralytics
+python -m venv .venv
+# Linux/macOS: source .venv/bin/activate
+# Windows PowerShell: .venv\Scripts\Activate.ps1
+python -m pip install -U pip
+python -m pip install -e ".[dev]"
```
Comment (UltralyticsAssistant) [inline RESEARCH_QUICKSTART.zh-CN.md]: 💡 **MEDIUM**: 后文第 319 行调用 ONNX 导出，但这里仅安装 `[dev]`。ONNX 导出所需的 `onnx`、`onnxslim` 和 `onnxruntime` 位于 `export-base` extra 中，因此按本节从全新环境执行会在导出阶段因缺少依赖失败。

**Suggested change:**
```suggestion
python -m pip install -e ".[dev,export-base]"
```

File: RESEARCH_QUICKSTART.zh-CN.md [hunk 3]
```diff
@@ -0,0 +1,364 @@
...
+
+from ultralytics import YOLO
+
+model = YOLO("path/to/my_yolo.yaml", task="detect", verbose=True)
+model.info()
+output = model.model(torch.zeros(1, 3, 640, 640))
+print(type(output))
+```
+
+`model.info()` 显示 0 FLOPs 通常意味着前向传播失败。先解决通道数、空间尺寸和 `from` 索引问题，再开始训练。
```
Comment (UltralyticsAssistant) [inline RESEARCH_QUICKSTART.zh-CN.md]: 📝 **LOW**: `model.info()` 的 FLOPs 由 `get_flops()` 计算，该函数在 `thop` 未安装或 profiling 抛异常时都会返回 0，即使单独前向传播成功；自定义算子尤其可能触发这种情况。因此将 0 FLOPs 通常解释为前向失败会误导排查，应说明这也可能表示 THOP 不可用或不支持该算子。

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Reviewed both changed Markdown files and traced the documented commands through the repository’s model, trainer, DDP, dataset, and export paths. The guide is generally accurate, but the custom-trainer example needs a multi-GPU importability caveat, the installation command must include export dependencies for the advertised ONNX workflow, and the FLOPs troubleshooting statement is too strong.

<details><summary>💬 Posted 3 inline comments</summary>

- 💡 **MEDIUM** `RESEARCH_QUICKSTART.zh-CN.md:221` 此示例把 `ResearchTrainer` 定义在调用脚本的 `__main__` 中。单卡可用，但多卡训练时 `BaseTrainer` 会生成临时 DDP 脚本，并按 `trainer.__class__.__module__` 执行 `from __main__ import ResearchTrainer`；此时 `__main__` 已是临时脚本，导入会失败。请将 trainer 放入可导入的模块后再传入，或明确该示例仅适用于单卡，并在多卡排查项中说明。
- 💡 **MEDIUM** `RESEARCH_QUICKSTART.zh-CN.md:239` 后文第 319 行调用 ONNX 导出，但这里仅安装 `[dev]`。ONNX 导出所需的 `onnx`、`onnxslim` 和 `onnxruntime` 位于 `export-base` extra 中，因此按本节从全新环境执行会在导出阶段因缺少依赖失败。
- 📝 **LOW** `RESEARCH_QUICKSTART.zh-CN.md:164` `model.info()` 的 FLOPs 由 `get_flops()` 计算，该函数在 `thop` 未安装或 profiling 抛异常时都会返回 0，即使单独前向传播成功；自定义算子尤其可能触发这种情况。因此将 0 FLOPs 通常解释为前向失败会误导排查，应说明这也可能表示 THOP 不可用或不支持该算子。
</details>

File: RESEARCH_QUICKSTART.zh-CN.md (hunk 1 above)
Comment (wangjia15) [reply RESEARCH_QUICKSTART.zh-CN.md:229]: 已修正：将 `ResearchTrainer` 移入独立的 `research_trainer.py`，训练脚本通过模块导入，并补充了 DDP 临时脚本依赖 `trainer.__class__.__module__` 可导入性的说明。

File: RESEARCH_QUICKSTART.zh-CN.md (hunk 2 above)
Comment (wangjia15) [reply RESEARCH_QUICKSTART.zh-CN.md]: 已采用建议，将 editable 安装更新为 `.[dev,export-base]`，使后文 ONNX 导出在全新环境中具备所需依赖。

File: RESEARCH_QUICKSTART.zh-CN.md (hunk 3 above)
Comment (wangjia15) [reply RESEARCH_QUICKSTART.zh-CN.md]: 已修正：现在明确 0 FLOPs 也可能由 THOP 缺失或不支持自定义算子导致，并建议用直接随机输入前向区分。

**Comment by pderrenger** [comment]
Thanks for addressing the review comments. The remaining blocker is the CLA check; please post `I have read the CLA Document and I sign the CLA` as a new comment to sign it.

**Comment by raimbekovm** [comment]
Thanks for the PR and for the care that went into it. I'm closing it because most of this content is already covered in the docs, and every page is available in Chinese through the site's language switcher: the YAML layer format, custom module registration, and troubleshooting are in the [模型 YAML 配置指南](https://docs.ultralytics.com/zh/guides/model-yaml-config/); custom trainers and callbacks are in [高级定制](https://docs.ultralytics.com/zh/usage/engine/), [自定义训练器](https://docs.ultralytics.com/zh/guides/custom-trainer/), and [回调](https://docs.ultralytics.com/zh/usage/callbacks/); the editable install is in the [快速入门](https://docs.ultralytics.com/zh/quickstart/); dataset YAML layout is in [目标检测数据集](https://docs.ultralytics.com/zh/datasets/detect/); and the train, val, and export workflow is on the [训练](https://docs.ultralytics.com/zh/modes/train/), [验证](https://docs.ultralytics.com/zh/modes/val/), and [导出](https://docs.ultralytics.com/zh/modes/export/) pages. A standalone file in the repo root would not be indexed by the docs site and would drift from these pages over time. Thanks again for contributing.
